pkg/webui/components/offline-status/index.js

```diff
@@ -44,34 +46,45 @@ const handleMessage = (message, type) => {
   })
 }
 
-const OfflineStatus = ({ showOfflineOnly, showWarnings, isOnline }) => {
+const OfflineStatus = ({ showOfflineOnly, showWarnings, onlineStatus }) => {
   const initialUpdate = useRef(true)
+  const isOnline = onlineStatus === ONLINE_STATUS.ONLINE
+  const isOffline = onlineStatus === ONLINE_STATUS.OFFLINE
+  const isChecking = onlineStatus === ONLINE_STATUS.CHECKING
+
   useEffect(() => {
     if (initialUpdate.current) {
       initialUpdate.current = false
       return
     }
     if (showWarnings && isOnline) {
       handleMessage(m.online, toast.types.INFO)
-    } else if (showWarnings && !isOnline) {
+    } else if (showWarnings && isOffline) {
       handleMessage(m.offline, toast.types.ERROR)
     }
-  }, [showWarnings, isOnline])
+  }, [showWarnings, isOnline, isOffline])
 
   if (showOfflineOnly && isOnline) {
     return null
   }
 
+  const icon = isOnline ? 'info' : isChecking ? 'warning' : 'error'
+  const message = isOnline
+    ? sharedMessages.online
+    : isChecking
+    ? sharedMessages.connectionIssues
+    : sharedMessages.offline
+
   return (
-    <span className={classnames(style.status, { [style.online]: isOnline })}>
-      <Icon className={style.icon} icon={isOnline ? 'info' : 'error'} />
-      <Message content={isOnline ? sharedMessages.online : sharedMessages.offline} />
+    <span className={classnames(style.status, style[onlineStatus])}>
```

Comment:
Using `style[onlineStatus]` can cause styling issues later, especially that there no tests for this component. If anyone changes values of `ONLINE_STATUS` this breaks.

I would prefer using something like this:
```js
{
  [style.online]: isOnline,
  [style.offline]: isOffline,
  [style.checking]: isChecking,
}